### PR TITLE
feat: Added generic api error ignore routine

### LINKED ISSUE #445: Add a generic handler of not found errors
### Describe the bug

Different resources has a different `NotFound` status. not found is not an error that should be returned.

### Expected Behavior

don't return errors if resource is not found

### Steps to Reproduce

Run describe ecs cluster on removed cluster

### Possible Solution

Create a generic handler that checks if error is related to not found.

### Provider and CloudQuery version

0.19.2; 0.9.2

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by amanenk**
Most of the errors that should be ignored has a few common prefixes/suffixes. The function checks if error should be ignored and if it is it logs error as warning and allows to skip it. The function is used only in a few places. After PR approve I will propagate it to other resources.
closes #445

### CODE REVIEW

**Comment by disq** [comment]
also: https://github.com/cloudquery/cq-provider-aws/pull/462

File: client/helpers.go [hunk 1]
```diff
@@ -218,3 +219,28 @@ func ResolveARN(service AWSService, resourceID func(resource *schema.Resource) (
...
 }
+
+func stringOneOf(a string, args ...interface{}) bool {
+	for _, b := range args {
+		if b == a {
+			return true
+		}
+	}
+	return false
+}
```
Comment (roneli) [inline client/helpers.go:230]: use the funk library instead you can use funk.Contains

File: client/helpers.go [hunk 2]
```diff
@@ -218,3 +219,28 @@ func ResolveARN(service AWSService, resourceID func(resource *schema.Resource) (
...
+
+// IgnoreApiError checks if api error should be ignored
+func (c *Client) IgnoreApiError(err error) bool {
```
Comment (roneli) [inline client/helpers.go:233]: the naming is wrong, its a IsNotFoundError

File: client/helpers.go [hunk 3]
```diff
@@ -218,3 +219,28 @@ func ResolveARN(service AWSService, resourceID func(resource *schema.Resource) (
...
+	var ae smithy.APIError
+	if errors.As(err, &ae) {
+		errorCode := ae.ErrorCode()
+		if stringOneOf(errorCode, "ResourceNotFoundException", "WAFNonexistentItemException") ||
+			strings.HasPrefix(errorCode, "NoSuch") ||
+			strings.HasSuffix(errorCode, "NotFound") ||
+			strings.HasSuffix(errorCode, "NotFoundError") {
```
Comment (roneli) [inline client/helpers.go:240]: we could use contains here no? or a regex?

File: client/helpers.go [hunk 4]
```diff
@@ -218,3 +219,28 @@ func ResolveARN(service AWSService, resourceID func(resource *schema.Resource) (
...
+	}
+	return false
+}
+
+// IgnoreApiError checks if api error should be ignored
+func (c *Client) IgnoreApiError(err error) bool {
+	var ae smithy.APIError
+	if errors.As(err, &ae) {
+		errorCode := ae.ErrorCode()
+		if stringOneOf(errorCode, "ResourceNotFoundException", "WAFNonexistentItemException") ||
```
Comment (roneli) [inline client/helpers.go:237]: would make this error codes a const of strings as it might grow larger overtime.

**Review by roneli — CHANGES_REQUESTED** [review]
Good helper method, had some comments

**Review by disq — APPROVED** [review]

File: client/helpers.go [hunk 5]
```diff
@@ -218,3 +218,26 @@ func ResolveARN(service AWSService, resourceID func(resource *schema.Resource) (
...
+	if errors.As(err, &ae) {
+		errorCode := ae.ErrorCode()
+		for _, s := range notFoundErrorPrefixes {
+			if strings.Contains(errorCode, s) {
+				c.logger.Warn("API returned \"NotFound\" error ignoring it...", "error", err)
+				return true
+			}
+		}
+	}
+	return false
```
Comment (roneli) [inline client/helpers.go:242]: Nit:
```suggestion
	if !errors.As(err, &ae) {
	   return false
	}
        errorCode := ae.ErrorCode()
        for _, s := range notFoundErrorPrefixes {
	        if strings.Contains(errorCode, s) {
		        c.logger.Warn("API returned \"NotFound\" error ignoring it...", "error", err)
		        return true
	        }
        }
        return false
```

**Review by roneli — APPROVED** [review]
LGTM, had one nit.

**Review by roneli — APPROVED** [review]
LGTM

**Pull request merged**